### ISSUE TITLE
Combining path and magick arguments outputs the wrong path

### ISSUE THREAD

**Issue by cakeisalie5** [envygeeks/jekyll-assets#491]
- [x] I tried updating to the latest version. (_I'm using the git source in my [Gemfile](https://forge.touhey.fr/teapots.git/tree/Gemfile#n21)_)
  - [ ] I can't, there is an issue.
- [ ] I Am on Windows
- [ ] I Am on Bash on Windows
- [x] I Am on Linux
- [ ] I Am on OS X

## Description
For a presentation website, I resize and crop images to 800x450 (16:9 ratio) then get the path to make a clickable image for which I control the attributes (probably only Pavlovian conditioning from the time jekyll-assets produced a double `alt` attribute, but that's not the point).

The code for resizing/cropping/getting the image path I use is the following (all variables are generated before, full code is [here](https://forge.touhey.fr/teapots.git/tree/_layouts/home.html#n23), I don't want to use Ruby instead yet):

```liquid
{% capture path
%}{% asset "{{ i.path }}" @magick:resize:"{{ w }}"x"{{ h }}" @magick:crop:"{{ pw }}"x"{{ ph }}"+"{{ cx }}"+"{{ cy }}" @path
%}{% endcapture
%}
```

The problem is that, in the generated code, the path doesn't lead to the resized & cropped image, but to the unedited (e.g. 3456x5184 pixels tall) original image (digest included). I tried removing the `@path` argument to check if the resizing and cropping arguments are correct, and the generated image had the expected result!

Thanks for your great project (although a little more documentation would be great!), and sorry for the inconvenience. :)

**Comment by envygeeks** [comment]
You are doing arguments the 2.x way, not the 3.x way: https://github.com/envygeeks/jekyll-assets#arguments.  I should also note that you are doing them rather confusingly, we parse the same way your CLI does in Bash, you don't need a dozen quotes, that also makes things confusing for you and your developers you can `"{{ w }}x{{ h }}"` instead of `"{{ w }}"x"{{ h }}"`, as by the time we get done parsing with `liquid-tag-parser`, the example I gave you is what it is anyways, so there is no need to make it harder on yourself to read, by the time we get done parsing the non-positional quotes into arguments it comes out just like bash.  So `"{{ w }}"x"{{ h }}"` is returned to us as a single string `{{ w }}x{{ h }}`.

Please update your arguments to be 3.x if you are on 3.x, if you continue to have problems, please do update this ticket and I'll happily look into it further, however at this time I'm closing this as PEBKAC (no offense intended that's just what we call users issues.)

**Comment by envygeeks** [comment]
```html
{% capture path %}
  {% asset "{{ i.path }}" @magick:resize="{{ w }}x{{ h }}" @magick:crop="{{ pw }}x{{ ph }}+{{ cx }}+{{ cy }}" @path %}
{% endcapture %}
```

**Comment by cakeisalie5** [comment]
Oh, I see; thanks, this syntax is indeed better! I've corrected the syntax (and a minor bug aside, no idea how it did work without the `@path` before, but the values have sense now) and tried rebuilding again, but I get the same result: I get the path of the unedited image (with the hash) instead of the resized/cropped one :/

**Comment by cakeisalie5** [comment]
Hi, sorry to insist but I can still reproduce the issue…

**Comment by cakeisalie5** [comment]
To answer to the “unreproducible” tag, the source of the website I have the issue on is still open [here](https://forge.touhey.fr/teapots.git/). It pulls the latest source from git, and uses the `vendor/` folder (if you use the Makefile command `make preview`, which runs Jekyll in watch mode), which can maybe help you for debugging (although I don't really know?).

I'm really really sorry for the inconvenience :/

**Comment by envygeeks** [comment]
Your issue is not being ignored, please test with v3.0.7 and see if it works now.  There was a recent issue with proxies fixed.  If that fixes your issue please do ping so this ticket can be reflagged and closed.  Otherwise it'll get fixed in v3.0.8 in the next few weeks.

Thanks!

**Comment by cakeisalie5** [comment]
After a `bundle update` (which updated the version of `jekyll-assets` to `v3.0.7`), I have tried with the `@path` before and after the magick filters and I still have the issue (the obtained images are not the modified ones but the original ones). But I believe the images are still processed (as before), only the URL/path is not the right one!

(also, but this issue is probably not the place to write this, an `asset_path` in my `.scss` file gives a path which looks valid, `/assets/head_blur-ca2dcbcbfd1b1216239962fd5f423fbf100bd239c6572d2068a1bd2f6f6858c0.jpg`, but results in a 404 and I could find no file starting with `head_blur`, should I open another issue about this?)

Thanks for your time :)

**Comment by envygeeks** [comment]
Yeah I know where this bug is. While adding a feature for a client I did an audit of tag, and hunted the bug pretty easily based off the guess that I was pretty dumb and over optimized and skipped proxies entirely if @path is shipped.  This'll be fixed in the next release, but be on master sometime today.

**Comment by envygeeks** [comment]
Okay, I've added test cases, confirmed it's not a bug, so I re-read your ticket, this bug is probably caused by your usage, and I didn't even notice at first, which means I need to add more documentation.

You should actually be doing:

```
{% capture path %}
  {% asset "{{ i.path }}" magick:resize="{{ w }}x{{ h }}" magick:crop="{{ pw }}x{{ ph }}+{{ cx }}+{{ cy }}" @path %}
{% endcapture %}
```

The `@` syntax marks a boolean, it's not a marker past that, other arguments with values should be just "key:value" for a proxy, or "key" otherwise.  "@key:val=whatever" will just evaluate to true, because our parser ignores you value and makes it true, as per the spec.

**Comment by envygeeks** [comment]
But there might be another bug elsewhere that needs to be dealt with, I'll update or close this bug soon based on further research and huting, and once I know more about the case in whole.

**Comment by envygeeks** [comment]
So I've confirmed on my end this is not a bug, but in the process of testing this and discovering it was a syntax problem, I did discover another bug, so thanks for this report as it prompted me to audit my source: https://github.com/envygeeks/jekyll-assets/commit/9b087f21373490d439c4ea6657a68850a220a367

I'm gonna close this for now, as 3.0.8 is going out soon, please do test it and confirm!

**Comment by cakeisalie5** [comment]
The operations are indeed combined, I get the path and the image is cropped and resized! :)

But… the operations do not happen in the right order. I've tried with the order you gave me, and with resizing and cropping reversed, it seems that the cropping always happens first, where in the order I wrote, the resizing should happen first… :(

**Comment by envygeeks** [comment]
I don't allow the "ordering" of any proxy, in any capacity. Our arguments are unordered, and nonpositional. What you want is a feature request, there are no promises it won't be requested to be sponsored though (by anybody.)

**Comment by cakeisalie5** [comment]
I see. Well, thanks a lot for your time and help anyway! :)
